src/renderer/viz/expressions/basic/List.js

```diff
@@ -0,0 +1,78 @@
+import Base from '../base';
+import { checkExpression, implicitCast, getOrdinalFromIndex, checkMaxArguments } from '../utils';
+import ListImage from '../ListImage';
+import ListGeneric from './ListGeneric';
+
+const SUPPORTED_CHILD_TYPES = ['number', 'category', 'color', 'time', 'image'];
+
+/**
+ * Wrapper around arrays. Explicit usage is unnecessary since CARTO VL will wrap implicitly all arrays using this function.
+ *
+ * @param {Number[]|Category[]|Color[]|Date[]|Image[]} elements
+ * @returns {List}
+ *
+ * @memberof carto.expressions
+ * @name array
+ * @function
+ * @api
+ */
+export default class List extends Base {
+    constructor (elems) {
+        checkMaxArguments(arguments, 1, 'list');
+
+        if (!elems) {
+            throw new Error('list(): invalid parameters: must receive at least one argument');
+        }
+
+        if (!window.Array.isArray(elems)) {
+            elems = [elems];
+        }
+
+        elems = elems.map(implicitCast);
+
+        if (!elems.length) {
+            throw new Error('list(): invalid parameters: must receive at least one argument');
+        }
+
+        elems.map((item, index) => {
+            checkExpression('list', `item[${index}]`, index, item);
+        });
+
+        super(elems);
+        this.elems = elems;
+        this._setTypes();
```

Comment:
this shouldn't be called in the constructor, types should be put on `_bindMetadata`